extensions/src/main/java/org/apache/falcon/extensions/mirroring/hdfsSnapshot/HdfsSnapshotMirroringExtension.java

```diff
@@ -0,0 +1,239 @@
+            LOG.debug("HDFS Snapshot extension validating if dir {} is snapshotable.", path.toString());
+            SnapshottableDirectoryStatus[] snapshotableDirs = hdfs.getSnapshottableDirListing();
+            if (snapshotableDirs != null && snapshotableDirs.length > 0) {
+                for (SnapshottableDirectoryStatus dir : snapshotableDirs) {
+                    if (dir.getFullPath().toString().equals(path.toString())) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        } catch (IOException e) {
+            LOG.error("Unable to verify if dir {} is snapshot-able. {}", path.toString(), e.getMessage());
+            throw new FalconException("Unable to verify if dir " + path.toString() + " is snapshot-able", e);
+        }
+    }
+
+    @Override
+    public Properties getAdditionalProperties(final Properties extensionProperties) throws FalconException {
+        Properties additionalProperties = new Properties();
+
+        // Add default properties if not passed
+        String distcpMaxMaps = extensionProperties.getProperty(HdfsSnapshotMirrorProperties.DISTCP_MAX_MAPS.getName());
+        if (StringUtils.isBlank(distcpMaxMaps)) {
+            additionalProperties.put(HdfsSnapshotMirrorProperties.DISTCP_MAX_MAPS.getName(), "1");
+        }
+
+        String distcpMapBandwidth = extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.MAP_BANDWIDTH_IN_MB.getName());
+        if (StringUtils.isBlank(distcpMapBandwidth)) {
+            additionalProperties.put(HdfsSnapshotMirrorProperties.MAP_BANDWIDTH_IN_MB.getName(), "100");
+        }
+
+        String tdeEnabled = extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.TDE_ENCRYPTION_ENABLED.getName());
+        if (StringUtils.isBlank(tdeEnabled)) {
+            additionalProperties.put(HdfsSnapshotMirrorProperties.TDE_ENCRYPTION_ENABLED.getName(), "false");
+        }
+
+        // Add sourceCluster properties
+        Cluster sourceCluster = ClusterHelper.getCluster(extensionProperties.getProperty(
+                HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName()));
+        if (sourceCluster == null) {
+            LOG.error("Cluster entity {} not found", HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName());
+            throw new FalconException("Cluster entity "
+                    + HdfsSnapshotMirrorProperties.SOURCE_CLUSTER.getName() + " not found");
+        }
+        additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_NN.getName(),
+                ClusterHelper.getStorageUrl(sourceCluster));
+        additionalProperties.put(HdfsSnapshotMirrorProperties.SOURCE_EXEC_URL.getName(),
+                ClusterHelper.getMREndPoint(sourceCluster));
+        String sourceKerberosPrincipal = ClusterHelper.getPropertyValue(sourceCluster, SecurityUtil.NN_PRINCIPAL);
+        if (StringUtils.isNotBlank(sourceKerberosPrincipal)) {
```

Comment:
Please get all the security related properties if security is enabled instead of checking for null